sdk/resources/Azure.Management.Resources/tests/InMemoryTests/DeploymentTests.InMemory.cs

```diff
@@ -0,0 +1,967 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.TestFramework;
+using Azure.Management.Resources;
+using Azure.Management.Resources.Models;
+using Azure.Management.Resources.Tests;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+
+namespace ResourceGroups.Tests
+{
+    public class InMemoryDeploymentTests : ClientTestBase
+    {
+        public InMemoryDeploymentTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        public ResourcesManagementClient GetResourceManagementClient(HttpPipelineTransport transport)
+        {
+            ResourcesManagementClientOptions options = new ResourcesManagementClientOptions();
+            options.Transport = transport;
+
+            return InstrumentClient(new ResourcesManagementClient(
+                "ddd5f1d5-499c-45ee-a5ff-024d36ade650",
```

Comment:
we should use TestEnvironment.SubscriptionId instead of hard code

Reply:
Done